apps/ensrainbow/src/utils/env-utils.ts

```diff
@@ -0,0 +1,298 @@
+}
+
+/**
+ * Parses a string value from an environment variable
+ *
+ * @param envVarName The name of the environment variable
+ * @param defaultValue Optional default value if the environment variable is not set
+ * @param validator Optional validation function that should return:
+ *                  - true if validation passes
+ *                  - false if validation fails (a generic error message will be used)
+ *                  - a string with a custom error message if validation fails
+ * @returns The parsed string value
+ * @throws Error if validation fails or if the environment variable is not set and no default is provided
+ */
+export const getEnvString = (
+  envVarName: string,
+  defaultValue?: string,
+  validator?: (value: string) => boolean | string,
+): string => {
+  try {
+    const rawValue = process.env[envVarName];
+
+    // if no value is provided, return the default or throw an error
+    if (rawValue === undefined) {
+      if (defaultValue !== undefined) {
+        return defaultValue;
+      }
+      throw new Error(
+        `Environment variable '${envVarName}' is not set and no default value was provided.`,
+      );
+    }
+
+    // if a validator is provided, run it
+    if (validator) {
+      const validationResult = validator(rawValue);
+      if (validationResult !== true) {
+        const errorMessage =
+          typeof validationResult === "string"
+            ? validationResult
+            : `Invalid value '${rawValue}' for environment variable '${envVarName}'.`;
+        throw new Error(errorMessage);
+      }
+    }
+
+    return rawValue;
+  } catch (error) {
+    const errorMessage = formatEnvVarError(envVarName, error);
+    throw new Error(errorMessage);
+  }
+};
+
+/**
+ * Parses a non-negative integer from an environment variable
+ *
+ * @param envVarName The name of the environment variable
+ * @param defaultValue Optional default value if the environment variable is not set
+ * @returns The parsed non-negative integer
+ * @throws Error if the value is not a valid non-negative integer
+ */
+export const getEnvNonNegativeInteger = (envVarName: string, defaultValue?: number): number => {
+  try {
+    const rawValue = process.env[envVarName];
+
+    // if no value is provided, return the default or throw an error
+    if (rawValue === undefined) {
+      if (defaultValue !== undefined) {
+        return defaultValue;
+      }
+      throw new Error(
+        `Environment variable '${envVarName}' is not set and no default value was provided.`,
+      );
+    }
+
+    try {
+      return parseNonNegativeInteger(rawValue);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        throw new Error(`Invalid value for environment variable '${envVarName}': ${error.message}`);
+      }
+      throw error;
+    }
+  } catch (error) {
+    const errorMessage = formatEnvVarError(envVarName, error);
+    throw new Error(errorMessage);
+  }
```

Comment:
`formatEnvVarError` handles error type conversion to string, so we could flatten the try/catch structure:
```suggestion
  try {
    const rawValue = process.env[envVarName];

    // if no value is provided, return the default or throw an error
    if (rawValue === undefined) {
      if (defaultValue !== undefined) {
        return defaultValue;
      }
      throw new Error(
        `Environment variable '${envVarName}' is not set and no default value was provided.`,
      );
    }

    return parseNonNegativeInteger(rawValue);
  } catch (error) {
    const errorMessage = formatEnvVarError(envVarName, error);
    throw new Error(errorMessage);
  }
```

Reply:
That's right, thanks for the catch